### PR TITLE
Add firebase support

### PR DESCRIPTION
**Pull request by manoldonev**
Follows closely the TypeScript implemenation.
Also added custom eslint ruleset.

### CODE REVIEW

File: cars/cars-list-view-model.js [hunk 1]
```diff
@@ -1,43 +1,42 @@
-var config = require("../shared/config");
-var fetchModule = require("fetch");
-var Observable = require("data/observable").Observable;
-var ObservableArray = require("data/observable-array").ObservableArray;
+
```
Comment (MIYankov) [inline cars/cars-list-view-model.js]: Empty line on top.

File: cars/cars-list-view-model.js [hunk 2]
```diff
@@ -1,43 +1,42 @@
...
-                viewModel.set("isLoading", false);
-            });
-    }
-
-    viewModel.empty = function () {
+        const onValueEvent = (snapshot) => {
+            this._handleSnapshot(snapshot.value);
+            this.set("isLoading", false);
+        };
+        firebase.addValueEventListener(onValueEvent, `/${path}`);
```
Comment (MIYankov) [inline cars/cars-list-view-model.js]: Shouldn't we have a service like in the angular template?

File: cars/shared/car-model.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const Observable = require("data/observable").Observable;
+
+function Car(options) {
+    const viewModel = new Observable();
```
Comment (MIYankov) [inline cars/shared/car-model.js]: I might be wrong, but I think this doesn't need to be observable nor a view model.

File: cars/car-detail-edit-page/list-selector/list-selector-modal-page.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const ListSelectorViewModel = require("./list-selector-view-model");
+
+let viewModel;
```
Comment (MIYankov) [inline cars/car-detail-edit-page/list-selector/list-selector-modal-page.js]: We should discus how we handle the reuse of the view model. In other templates we get the view model from args.object.bindingContext.

File: cars/car-detail-edit-page/list-selector/list-selector-view-model.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+const ObservableArray = require("data/observable-array").ObservableArray;
+const { carClassList, carDoorList, carSeatList, carTransmissionList } = require("./constants");
+
+function ListSelectorViewModel(context, closeCallback) {
+    const viewModel = new Observable();
+
+    viewModel._closeCallback = closeCallback;
+    viewModel.title = `Select Car ${capitalizeFirstLetter(context.tag)}`;
+
+    const protoItems = resolveProtoItems(context.tag);
```
Comment (MIYankov) [inline cars/car-detail-edit-page/list-selector/list-selector-view-model.js]: Maybe it's just me, but I don't get what proto items are. It's just the naming I think. Plus, they are not named in the angular template.

File: cars/cars-list-view-model.js (hunk 2 above)
Comment (manoldonev) [inline cars/cars-list-view-model.js]: Not sure, I'll have to think about it -- we can add this as an enhancement later.

File: cars/shared/car-model.js (hunk 3 above)
Comment (manoldonev) [inline cars/shared/car-model.js]: If this is not observable, the slider value two-way binding does not work on car-detail-edit-page.xml:

```
<Slider minValue="{{ carLuggageMinValue }}" maxValue="{{ carLuggageMaxValue }}" value="{{ car.luggage }}" />
```

File: cars/car-detail-edit-page/list-selector/list-selector-view-model.js (hunk 5 above)
Comment (manoldonev) [inline cars/car-detail-edit-page/list-selector/list-selector-view-model.js]: proto-
/ˈprəʊtəʊ/

original or primitive.
"prototherian"
first or earliest.
"protomartyr"

Do you have better suggestions?

File: cars/car-detail-edit-page/list-selector/list-selector-modal-page.js (hunk 4 above)
Comment (manoldonev) [inline cars/car-detail-edit-page/list-selector/list-selector-modal-page.js]: Ok, but I am not reusing ListSelectorViewModel here?

**Pull request merged**